src/socketio.quiver.js

```diff
@@ -144,11 +144,48 @@ QuiverSocialProvider.configuration_ = undefined;
  */
 QuiverSocialProvider.connection_ = undefined;
 
+/**
+ * @param {!Array<T>} list To sample from
+ * @param {number} k Target number of samples
+ * @return {!Array<T>} Subset of |list|
+ * @template T
+ */
+QuiverSocialProvider.randomSample_ = function(list, k) {
+  if (k >= list.length) {
+    return list;
+  }
+
+  // Algorithm R: https://en.wikipedia.org/wiki/Reservoir_sampling#Algorithm_R
+  var sample = list.slice(0, k);
+  for (var i = k; i < list.length; ++i) {
+    var j = Math.floor(Math.random() * i);
```

Comment:
In general it looks like with k==2, list[0] and list[1] are 50% less likely to be chosen then later elements.  I have some notes and can explain this for a simple case where list.length==4.  Maybe the algorithm we want is https://en.wikipedia.org/wiki/Fisher–Yates_shuffle?  Or we could use _.shuffle (which implements that algorithm)